### PR TITLE
Add support for specifying AZ in VPN Gateway

### PR DESCRIPTION
**Pull request by DrFaust92**
# Description

add support for specifying AZ in VPN Gateway.

unfortunately i cannot test this.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1020,8 +1020,9 @@ resource "aws_customer_gateway" "this" {
 resource "aws_vpn_gateway" "this" {
   count = var.create_vpc && var.enable_vpn_gateway ? 1 : 0
 
-  vpc_id          = local.vpc_id
-  amazon_side_asn = var.amazon_side_asn
+  vpc_id            = local.vpc_id
+  amazon_side_asn   = var.amazon_side_asn
+  availability_zone = length(var.vpn_gateway_az) != 0 ? var.vpn_gateway_az : null
```
Comment (antonbabenko) [inline main.tf]: Simpler => `availability_zone  = var.vpn_gateway_az`

File: variables.tf [hunk 2]
```diff
@@ -1327,6 +1327,11 @@ variable "amazon_side_asn" {
   default     = "64512"
 }
 
+variable "vpn_gateway_az" {
+  description = "The Availability Zone for the VPN Gateway"
+  default     = ""
```
Comment (antonbabenko) [inline variables.tf]:
```
type = string
default = null
```

**Comment by DrFaust92** [comment]
Fixed

**Comment by antonbabenko** [comment]
Thanks! [v2.28.0](https://github.com/terraform-aws-modules/terraform-aws-vpc/releases/v2.28.0) has been just released.

**Pull request merged**